row.go

```diff
@@ -521,14 +530,26 @@ func reconstructFuncOf(columnIndex int16, node Node) (int16, reconstructFunc) {
 
 //go:noinline
 func reconstructFuncOfOptional(columnIndex int16, node Node) (int16, reconstructFunc) {
+	// What's the purpose of forcing the passed in node to Required for the Optional fields?
+	// Ah, I see it's so that we eventually reach the leaf base-case?
```

Comment:
Let's confirm and add a real comment here, since this is confusing.

Reply:
The intent is to avoid infinite recursion, otherwise the `reconstructFuncOf` call below would call `reconstructFuncOfOptional` again.